asdf/lazy_nodes.py

```diff
@@ -109,7 +110,11 @@ def _to_lazy_node(node, af_ref):
     If the object does not have a corresponding subclass
     it will be returned unchanged.
     """
-    if isinstance(node, list):
+    if isinstance(node, tagged.TaggedList):
```

Comment:
I'm going to investigate this further but I think there may be a related bug where a unknown tag can cause more distal tree portions to go unconverted for a lazy tree. I may roll that fix into this PR (since it was revealed by the added tests). More investigation to follow.

Reply:
See https://github.com/asdf-format/asdf/pull/1983/commits/c7b1c18803cfc0a1ae9cd9409c58f4aa2f242044 for an updated unit test. In brief, when an unknown tag was encountered for a lazy tree it resulted in:
- loss of the tag
- return of the non-tagged dict/list

With this PR the result is a `Tagged` object that:
- keeps the tag
- for `TaggedDict` and `TaggedList` the backing dict/list is lazy